src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -674,87 +675,93 @@ private Index HandleStaticIndexChange(string name, IndexDefinition definition)
-                                break;
-
-                            default:
-                                throw new NotSupportedException($"Cannot create {definition.Type} index from IndexDefinition");
-                        }
-
-                        break;
-
-                    case IndexSourceType.TimeSeries:
-                        switch (definition.Type)
-                        {
-                            case IndexType.Map:
-                            case IndexType.JavaScriptMap:
-                                index = MapTimeSeriesIndex.CreateNew(definition, _documentDatabase);
-                                break;
-
-                            case IndexType.MapReduce:
-                            case IndexType.JavaScriptMapReduce:
-                                var mapReduceIndex = MapReduceIndex.CreateNew<MapReduceTimeSeriesIndex>(definition, _documentDatabase);
-
-                                if (mapReduceIndex.OutputReduceToCollection != null && prefixesOfDocumentsToDelete.Count > 0)
-                                    mapReduceIndex.OutputReduceToCollection.AddPrefixesOfDocumentsToDelete(prefixesOfDocumentsToDelete);
-
-                                index = mapReduceIndex;
-                                break;
-
-                            default:
-                                throw new NotSupportedException($"Cannot create {definition.Type} index from TimeSeriesIndexDefinition");
-                        }
+                return CreateIndexFromDefinition(definition, _documentDatabase, prefixesOfDocumentsToDelete: prefixesOfDocumentsToDelete);
+            }
+        }
 
-                        break;
+        public Index CreateTestIndexFromDefinition(IndexDefinition indexDefinition, DocumentDatabase documentDatabase, DocumentsOperationContext context)
+        {
+            if (indexDefinition.Type == IndexType.JavaScriptMapReduce || indexDefinition.Type == IndexType.MapReduce)
+                indexDefinition.OutputReduceToCollection = null;
+            
+            var testIndexConfiguration = new TestIndexConfiguration(indexDefinition.Configuration, documentDatabase.Configuration);
 
-                    case IndexSourceType.Counters:
-                        switch (definition.Type)
-                        {
-                            case IndexType.Map:
-                            case IndexType.JavaScriptMap:
-                                index = MapCountersIndex.CreateNew(definition, _documentDatabase);
-                                break;
+            var testIndex = CreateIndexFromDefinition(indexDefinition, documentDatabase, testIndexConfiguration);
 
-                            case IndexType.MapReduce:
-                            case IndexType.JavaScriptMapReduce:
-                                var mapReduceIndex = MapReduceIndex.CreateNew<MapReduceCountersIndex>(definition, _documentDatabase);
+            testIndex.InitializeTestIndex(context);
```

Comment:
Small readability issue - we have `testIndex. ... TestIndex`. Maybe let's rename the method to something like `InitializeTestRun()`

Reply:
Renamed